app/src/main/java/org/p2p/wallet/common/EncryptedSharedPreferences.kt

```diff
@@ -0,0 +1,255 @@
+package org.p2p.wallet.common
+
+import androidx.core.content.edit
+import android.content.SharedPreferences
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import timber.log.Timber
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty
+import org.p2p.core.crypto.Hex
+import org.p2p.wallet.common.crypto.keystore.DecodeCipher
+import org.p2p.wallet.common.crypto.keystore.EncodeCipher
+import org.p2p.wallet.common.crypto.keystore.KeyStoreWrapper
+
+class EncryptedSharedPreferences(
+    private val keyStoreWrapper: KeyStoreWrapper,
+    private val sharedPreferences: SharedPreferences,
+    private val gson: Gson
+) {
+    var loggingTag: String = "EncryptedSharedPreferences"
```

Comment:
should be const val

Reply:
and can be just TAG

Reply:
sorry, but no
you can write your own tag here for your storage.
i'll write a kdoc